src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java

```diff
@@ -131,24 +143,54 @@ public CollectionExercise findCollectionExercise(UUID id) {
     return linkedSummaries;
   }
 
+    /**
+     * Sets the values in a supplied collection exercise from a supplied DTO.
+     * WARNING: Mutates collection exercise
+     * @param collex the dto containing the data
+     * @param collectionExercise the collection exercise to apply the value from the dto to
+     */
+  private void setCollectionExerciseFromDto(CollectionExerciseDTO collex, CollectionExercise collectionExercise) {
+      collectionExercise.setName(collex.getName());
+      collectionExercise.setUserDescription(collex.getUserDescription());
+      collectionExercise.setExerciseRef(collex.getExerciseRef());
+      collectionExercise.setSurveyId(UUID.fromString(collex.getSurveyId()));
+
+      // In the strictest sense, some of these dates are mandatory fields for collection exercises.  However as they
+      // are not supplied at creation time, but later as "events" we will allow them to be null
+      if (collex.getScheduledStartDateTime() != null) {
+          collectionExercise.setScheduledStartDateTime(new Timestamp(collex.getScheduledStartDateTime().getTime()));
+      }
+      if (collex.getScheduledEndDateTime() != null) {
+          collectionExercise.setScheduledEndDateTime(new Timestamp(collex.getScheduledEndDateTime().getTime()));
+      }
+      if (collex.getScheduledExecutionDateTime() != null) {
+          collectionExercise.setScheduledExecutionDateTime(
+                  new Timestamp(collex.getScheduledExecutionDateTime().getTime()));
+      }
+      if (collex.getActualExecutionDateTime() != null) {
+          collectionExercise.setActualExecutionDateTime(new Timestamp(collex.getActualExecutionDateTime().getTime()));
+      }
```

Comment:
The `actionSvc` fails to retrieve collection exercise dates. Please refer a log file snippet below

`CollectionExerciseClientServiceImpl : cause = null - message = Can not deserialize value of type java.util.Date from String "2018-01-18T17:47:46.976Z": not a valid representation (error: Failed to parse Date value '2018-01-18T17:47:46.976Z': Unparseable date: "2018-01-18T17:47:46.976Z")
 at [Source: {"id":"14fb3e68-4dca-46db-bf49-04b84e07e77c","surveyId":"cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87","name":"BRES_2017","actualExecutionDateTime":"2018-01-18T17:47:46.976Z","scheduledExecutionDateTime":"2017-09-11T00:00:00.000+01:00","scheduledStartDateTime":"2017-09-12T00:00:00.000+01:00","actualPublishDateTime":"2018-01-18T16:24:20.868Z","periodStartDateTime":"2017-09-15T00:00:00.000+01:00","periodEndDateTime":"2017-09-15T23:59:59.000+01:00","scheduledReturnDateTime":"2017-10-06T01:00:00.000+01:00","scheduledEndDateTime":"2018-06-30T00:00:00.000+01:00","executedBy":null,"state":"VALIDATED","caseTypes":[{"actionPlanId":"e71002ac-3575-47eb-b87f-cd9db92bf9a7","sampleUnitType":"B"},{"actionPlanId":"0009e978-0932-463b-a2a1-b45cb3ffcb2a","sampleUnitType":"BI"}],"exerciseRef":"221_201712","userDescription":null,"created":null,"updated":null,"deleted":false}; line: 1, column: 141] (through reference chain: uk.gov.ons.ctp.response.collection.exercise.representation.CollectionExerciseDTO["actualExecutionDateTime"])`

Reply:
Strictly speaking it is the collection instrument service that is at fault here, as "2018-01-18T17:47:46.976Z" is a valid ISO date.  However I've come to realise that all the services show this same problem - I've fixed this strictness for collection exercise but this has clearly caused issues elsewhere where the date handling is incorrect